### PR TITLE
[BugFix] add fsync to hdfs-writer before close

### PR DESCRIPTION
**Pull request by xiaoyong-z**
## What type of PR is this：
- [x] bugfix
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #https://github.com/StarRocks/StarRocksTest/issues/981

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/fs/fs_hdfs.cpp [hunk 1]
```diff
@@ -174,11 +174,16 @@ Status HDFSWritableFile::close() {
         return Status::OK();
     }
     auto ret = call_hdfs_scan_function_in_pthread([this]() {
-        int r = hdfsCloseFile(this->_fs, this->_file);
+        int r = hdfsHSync(_fs, _file);
```
Comment (chaoyli) [inline be/src/fs/fs_hdfs.cpp:177]: You can add some commit message to describe the reason for the sync.

File: be/src/fs/fs_hdfs.cpp [hunk 2]
```diff
@@ -174,11 +174,16 @@ Status HDFSWritableFile::close() {
...
+        if (r != 0) {
+            return Status::IOError("sync error");
```
Comment (chaoyli) [inline be/src/fs/fs_hdfs.cpp:179]: Add the file name in the log.

**Review by chaoyli — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by xiaoyong-z** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
run starrocks_admit_test

**Pull request merged**